lib/keycloak.js

```diff
@@ -1102,26 +1093,21 @@ export default class Keycloak {
      * @returns {ParsedCallbackParams}
      */
     #parseCallbackParams(paramsString, supportedParams) {
-        var p = paramsString.split('&');
-
-        /** @type {ParsedCallbackParams} */
-        var result = {
-            paramsString: '',
-            oauthParams: {}
-        }
-        for (var i = 0; i < p.length; i++) {
-            var split = p[i].indexOf("=");
-            var key = p[i].slice(0, split);
-            if (supportedParams.indexOf(key) !== -1) {
-                result.oauthParams[key] = p[i].slice(split + 1);
-            } else {
-                if (result.paramsString !== '') {
-                    result.paramsString += '&';
-                }
-                result.paramsString += p[i];
+        const params = new URLSearchParams(paramsString);
+        /** @type {Record<string, string>} */
+        const oauthParams = {};
+
+        for (const [key, value] of Array.from(params.entries())) {
```

Comment:
I think the `Array.from()` is not needed and might be confusing. I think we can stick to:

```suggestion
        for (const [key, value] of params) {
```

Reply:
That is what I started out with, but apparently removing values while iterating causes some issues, which might lead to some elements remaining. Creating an array forces all values to be extracted right away, mitigating this problem.

Reply:
I see, thought it was implemented similar to Map, but apparently this is a documented pitfall:
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Iteration_protocols#concurrent_modifications_when_iterating.

TIL

Reply:
This surprised me as well, you learn something new every day. I will mark this as resolved.